packages/deployment/upgrade-test/upgrade-test-scripts/agoric-upgrade-11/zoe-full-upgrade/run-prober-script.js

```diff
@@ -0,0 +1,95 @@
+// to turn on ts-check:
+/* global E */
+
+// import { E } from "@endo/far";
+
+const PROBER_BUNDLE_ID =
+  'b1-4487fd5e18a225a77ca882890d8faae905ca46fda617877f6e1f969e624db3edf0a4f490d735963f025eb3d772840caec157986f4367c0f66385d919f3f5a56c';
+
+console.info('running zcf prober');
+
+const sub = (a, v) => {
+  return { brand: a.brand, value: a.value - v };
+};
+
+const probeReallocation = async (
+  value,
+  payment,
+  creatorFacet,
+  zoe,
+  ducatIssuer,
+) => {
+  const stagingInv = await E(creatorFacet).makeProbeStagingInvitation();
+
+  const stagingSeat = await E(zoe).offer(
+    stagingInv,
+    { give: { Ducats: value } },
+    { Ducats: payment },
+  );
+  const helperPayments = await E(stagingSeat).getPayouts();
+
+  const helperInv = await E(creatorFacet).makeProbeHelperInvitation();
+  const helperSeat = await E(zoe).offer(
+    helperInv,
+    { give: { Ducats: sub(value, 1n) } },
+    { Ducats: helperPayments.Ducats },
+  );
+  const internalPayments = await E(helperSeat).getPayouts();
+  const ducatAmount = await E(ducatIssuer).getAmountOf(internalPayments.Ducats);
+
+  const internalInv = await E(creatorFacet).makeProbeInternalInvitation();
+  const internalSeat = await E(zoe).offer(
+    internalInv,
+    { give: { Ducats: ducatAmount } },
+    { Ducats: internalPayments.Ducats },
+  );
+  const leftoverPayments = await E(internalSeat).getPayouts();
+
+  return {
+    stagingResult: await E(stagingSeat).getOfferResult(),
+    helperResult: await E(helperSeat).getOfferResult(),
+    internalResult: await E(internalSeat).getOfferResult(),
+    leftoverPayments,
+  };
+};
+
+/*
+ * Test a full upgrade of Zzoe and ZCF.
```

Comment:
💤

Reply:
Took me a couple of passes to see it. Thx.